js/src/BuilderSettingsMenu.js

```diff
@@ -0,0 +1,409 @@
+/**
+ * SettingsMenu
+ */
+
+/** @jsx h */
+import { h, Component } from 'preact'
+import '../../css/src/SettingsMenu.css'
+
+const ScaleEditor = require('./ScaleEditor')
+
+class BuilderSettingsMenu extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      display: props.display
+    }
+    props.settings.hold_changes()
+  }
+
+  componentWillReceiveProps (nextProps) {
+    this.setState({display: nextProps.display})
+    nextProps.settings.hold_changes()
+  }
+
+  abandonChanges () {
+    this.props.settings.abandon_changes()
+    this.setState({display: 'none'})
+  }
+
+  saveChanges () {
+    this.props.settings.accept_changes()
+    this.setState({display: 'none'})
+  }
+
+  handleStyle (value, type) {
+    let style = []
+    if (type === 'reaction_styles') {
+      style.concat(this.props.reaction_styles)
+    } else if (type === 'metabolite_styles') {
+      style.concat(this.props.metabolite_styles)
+    }
+    if (style) {
+      if (style.indexOf(value) > -1) {
+        style.splice(style.indexOf(value), 1)
+        this.props.settings.set_conditional(type, style)
```

Comment:
This line is a duplicate, let's put it after the conditional statement

Reply:
This code is no longer used